src/Validation/URLValidationRESTController.php

```diff
@@ -240,6 +313,32 @@ public function validate_post_url( $request ) {
 		return rest_ensure_response( $this->filter_response_by_context( $data, $request['context'] ) );
 	}
 
+	/**
+	 * Returns validation information about a URL.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_validated_url( $request ) {
+		$post_id       = (int) $request['id'];
+		$validated_url = new ValidatedUrlDataProvider( $post_id );
```

Comment:
I wasn't sure if it's the best practice to instantiate a `Provider` inside a method like here. Other providers are passed to the controller's `__construtor`. Or else, maybe the `ValidatedUrlDataProvider` shouldn't be a `Provider` in the first place?

Reply:
Any use of `new` for something that is not a newable object (i.e. value objects, factories, ...) is not best practice, as we're switching the code over to make use of an injector across the board.

As for your question about the nature of the provider in general, I think the modeling seems to be off.

Best practice would be to have the provider be injected via the constructor, and then have a method that accepts the post ID for which it should provide.
So, following from the name, I'd expect a `ValidatedUrlDataProvider` to provide a `ValidatedUrlData` object:
```php
$validated_url_data = $this->validated_url_data_provider->for( $post_id );
```
In your current design the provider _is_ the data it provides, as it receives the post ID already in its constructor. For multiple post IDs, you'd then need not multiple data objects, but multiple provider objects.

Hope this helps.

Reply:
Your feedback is very helpful, thanks @schlessera!

I've addressed the modelling problem in 5944bbc. According to your suggestion, there's a new `ValidatedUrlData` object that's returned by the provider.

The `ValidatedUrlData` object is a wrapper for the underlying `WP_Post` object. I'm wondering what would be the best model that would cover a potential data layer change from `WP_Post` to something else in the future.